runtime/lua/vim/treesitter/_fold.lua

```diff
@@ -261,10 +261,9 @@ end
 ---@param fn function
 local function schedule_if_loaded(bufnr, fn)
   vim.schedule(function()
-    if not api.nvim_buf_is_loaded(bufnr) then
-      return
+    if foldinfos[bufnr] and api.nvim_buf_is_loaded(bufnr) then
```

Comment:
Conceptually I think this should be done inside of `fn()`, so at the call sites to `schedule_if_loaded`.

Reply:
Both checks or just the newly added one?

Reply:
Just the new one. Otherwise this function isn't needed at all.

Reply:
Both checks are quite similar though, `foldinfos[bufnr]` is only `nil` _because_ the buffer was unloaded (but reloaded before `foldexpr()` is evaluated again). Doesn't it make sense to keep them together?

Reply:
Wrapping vim.schedule with a guard to check the buffer is loaded/valid is a common idiom I've used in a few plugins, so I'd imagine as we do more async this could eventually be generalized. `foldinfo` is state specific to this module, so IMO it makes sense to try and decouple that from this kind of logic. Not a big deal though.